projects/components/src/navigation/navigation-list.component.ts

```diff
@@ -95,6 +130,18 @@ export class NavigationListComponent implements OnChanges {
     return this.collapsed ? IconType.TriangleRight : IconType.TriangleLeft;
   }
 
+  public onNavItemSelected(navItemLink: NavItemLinkConfig): void {
+    if (this.usePageLevelTimeRange && !isNil(navItemLink.timeRange)) {
+      if (navItemLink.timeRange?.isCustom()) {
+        const timeRange: FixedTimeRange = navItemLink.timeRange;
+        this.timeRangeService.setFixedRange(timeRange.startTime, timeRange.endTime);
+      } else if (!navItemLink.timeRange?.isCustom()) {
+        const timeRange: RelativeTimeRange = navItemLink.timeRange as RelativeTimeRange;
+        this.timeRangeService.setRelativeRange(timeRange.duration.value, timeRange.duration.unit);
```

Comment:
So this is interesting - do we want to reset the time range every time we we navigate through the app? That can lead to inconsistent results, even if the time range doesn't change, and is a departure from what we do today. That may be OK at the top level navs like this (if not already done already, we should think through how we want it to behave if no change), but pretty sure this is not what we would want when following nav (I need to go find where that code is to see what we're doing there).

Update: Looks like following nav isn't changing, so that's good. It's just not being reassigned unless we click on one here. How's the initial load work? Does this get triggered by the initial active item?

Reply:
No this does not get triggered by initial active item. The initial load does not set the TR here, it would hit the default case in `time-range.service.ts`  in the init method.

Reply:
That sounds like a bug, right? From a behavior perspective we'd want the page-specific TR even on initial load, not just on clock.

Reply:
fixed. The TR is now set for the first active item on load.